Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,238 @@
+    let caseDescription = try values.decode(String.self, forKey: .caseDescription)
+    if caseDescription == Self.simpleDescription {
+      let text = try values.decode(String.self, forKey: .text)
+      self = .simple(name: text)
+
+    } else if caseDescription == Self.unknownDescription {
+      let text = try values.decode(String.self, forKey: .text)
+      self = .unknown(text: text)
+
+    } else if caseDescription == Self.memberDescription {
+      let text = try values.decode(String.self, forKey: .text)
+      let baseType = try values.decode(Self.self, forKey: .typeDescription)
+      self = .member(name: text, baseType: baseType)
+
+    } else if caseDescription == Self.optionalDescription {
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
+      self = .optional(typeDescription)
+
+    } else if caseDescription == Self.implicitlyUnwrappedDescription {
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
+      self = .implicitlyUnwrapped(typeDescription)
+
+    } else if caseDescription == Self.arrayDescription {
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
+      self = .array(typeDescription)
+
+    } else if caseDescription == Self.dictionaryDescription {
+      let key = try values.decode(Self.self, forKey: .typeDescriptionKey)
+      let value = try values.decode(Self.self, forKey: .typeDescriptionValue)
+      self = .dictionary(key: key, value: value)
+
+    } else if caseDescription == Self.compositionDescription {
+      let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
+      self = .composition(typeDescriptions)
+
+    } else if caseDescription == Self.tupleDescription {
+      let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
+      self = .tuple(typeDescriptions)
+
+    } else {
+      throw CodingError.unknownCase
+    }
+  }
+
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(caseDescription, forKey: .caseDescription)
+    switch self {
+    case let .simple(name):
+      try container.encode(name, forKey: .text)
+    case let .unknown(text):
+      try container.encode(text, forKey: .text)
+    case let .optional(type),
+         let .array(type),
+         let .implicitlyUnwrapped(type):
+      try container.encode(type, forKey: .typeDescription)
+    case let .tuple(types),
+         let .composition(types):
+      try container.encode(types, forKey: .typeDescriptions)
+    case let .dictionary(key, value):
+      try container.encode(key, forKey: .typeDescriptionKey)
+      try container.encode(value, forKey: .typeDescriptionValue)
+    case let .member(name, baseType):
+      try container.encode(name, forKey: .text)
+      try container.encode(baseType, forKey: .typeDescription)
+    }
+  }
+
+  enum CodingKeys: String, CodingKey {
```

Comment:
Documenting what these are intended to be used for would be helpful.

Reply:
6a2a8769b0f364cfc075e3378452abbf76c4e47b

Reply:
Very helpful!